packages/dag/src/group.ts

```diff
@@ -143,6 +145,46 @@ const buildPossiblyCyclicGroupGraph = <T>(
   return graph
 }
 
+const breakupDisjointGroup = <T>(
+  group: Group<T>,
+  source: DataNodeMap<T>,
+  groupKey: GroupKeyFunc,
+): GroupKeyFunc => {
+  const nodeIdToNewGroup = new Map<NodeId, string>()
+  const groupNodeIds = new Set(group.items.keys())
+  const groupGraph = wu(groupNodeIds)
+    .reduce((acc, nodeId) => {
+      acc.addNode(nodeId, intersection(source.get(nodeId), groupNodeIds), nodeId)
+      return acc
+    }, new DAG())
+
+  wu(groupGraph.evaluationOrderGroups(true))
+    .forEach(nodeIds => {
+      const groupId = _.uniqueId(`${group.groupKey}-`)
+      wu(nodeIds).forEach(nodeId => { nodeIdToNewGroup.set(nodeId, groupId) })
+    })
+
+  return (id: NodeId) => nodeIdToNewGroup.get(id) || groupKey(id)
+}
+
+const breakupDisjointGroups = <T>(
+  source: DataNodeMap<T>,
+  groupKey: GroupKeyFunc,
+  groups?: Set<NodeId>
+): GroupKeyFunc => {
+  if (_.isUndefined(groups) || _.isEmpty(groups)) {
```

Comment:
nit: I think `undefined` is considered empty.
so either
```suggestion
  if (groups === undefined || _.isEmpty(groups)) {
```
or
```suggestion
  if (_.isEmpty(groups)) {
```